src/core/interpreter.cc

```diff
@@ -694,16 +708,18 @@ int Interpreter::RedisGenericCommand(bool raise_error) {
         memcpy(cur, lua_tostring(lua_, idx), len);
     };
 
-    cmdargs[j] = {cur, len};
+    args[idx - 1] = {cur, len};
     cur += len;
   }
 
   /* Pop all arguments from the stack, we do not need them anymore
    * and this way we guaranty we will have room on the stack for the result. */
   lua_pop(lua_, argc);
   RedisTranslator translator(lua_);
-  redis_func_(MutSliceSpan{cmdargs}, &translator);
-  DCHECK_EQ(1, lua_gettop(lua_));
+  redis_func_(CallArgs{MutSliceSpan{args}, move(buffer), &translator, async});
```

Comment:
maybe instead of all the moves, fallback to old good c++ - 03?
Define  CallArgs on stack and pass to redis_func_ a mutable pointer to CallArgs?
I assume CallArgs are not accessed when we exit from RedisGenericCommand, right?

Reply:
There are only three moves in total, it shows clear intentions. I can revert it, but I don't really understand what the code issue is here